### PR TITLE
Draft of Templating Tutorial Post

### PR DESCRIPTION
**Pull request by kmpaul**
@jukent Take a look!

### CODE REVIEW

File: site/posts/templating-isnt-just-for-web-developers.md [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+
+This is exactly what "templating" is all about.  It is, essentially,
+just string substitution...but with a touch more sophistication.
+
+## The Jinja2 Template
+
+So, to start out, we are going to convert this text template into a
+Jinja2 template.  To do so, the easiest first step would be to just
+replace every `[]`-bracketted item with a Jinja2 *expressions*,
+which are repressented with `{{}}`-brackets:
```
Comment (andersy005) [inline site/posts/templating-isnt-just-for-web-developers.md:125]:
```suggestion
which are represented with `{{}}`-brackets:
```

File: site/posts/templating-isnt-just-for-web-developers.md [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+{{ closing }},
+
+{{ your_name }}
+{{ your_title }}
+```
+
+Also, notice that I have converted the text labels inside the
+`[]`-brackets into valid Python variable names.  Let's assume
+that this new *template file* is called `letter.j2` (where
+the `.j2` means it is a Jinja2 template).
```
Comment (andersy005) [inline site/posts/templating-isnt-just-for-web-developers.md:150]: The `.j2` file extension is not a must, right?

File: site/posts/templating-isnt-just-for-web-developers.md [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+held in the Bank of [Freedonia](https://en.wikipedia.org/wiki/Duck_Soup_(1933_film)).
+With tensions mounting between Freedonia and neighboring Sylvania,
+and war looming on the horizon, your client has authorized you to
+move these assets to another bank outside of Freedonia.  Clearly,
+your only course of action is to email as many people for whom
+you can find addresses to ask them if they would be willing to
+hold the assets in their account for the duration of the conflict.
+Obviously, you will need to pay them for their kind services, and
+the oil tycoon has agreed to authorize you to gift 20% of the assets
+to whomever holds them, as payment for their kindness.  You just
```
Comment (andersy005) [inline site/posts/templating-isnt-just-for-web-developers.md:76]: Where can I sign up? 😅

File: site/posts/templating-isnt-just-for-web-developers.md (hunk 2 above)
Comment (kmpaul) [reply site/posts/templating-isnt-just-for-web-developers.md:150]: Correct.  As far as I know, only the VS Code Jinja2 extension cares about the actual extension.  Jinja2 does not care at all.  I'll commend on that.

```suggestion
that this new *template file* is called `letter.j2` (where
the `.j2` means it is a Jinja2 template, but Jinja2 does not
care what extension the file actually has).
```

**Comment by kmpaul** [comment]
Thanks, @jukent!

**Pull request merged**